app-modules/project/src/Filament/Resources/ProjectResource/Pages/ListProjects.php

```diff
@@ -54,6 +56,21 @@ class ListProjects extends ListRecords
     public function table(Table $table): Table
     {
         return $table
+            ->modifyQueryUsing(function (Builder $query) {
+                $user = auth()->user();
+
+                if ($user->hasRole(Authenticatable::SUPER_ADMIN_ROLE)) {
+                    return $query;
+                }
+
+                return $query->where(function (Builder $query) use ($user) {
+                    $query->whereMorphedTo('createdBy', $user)
+                        ->orWhereHas('managerTeams.users', fn (Builder $query) => $query->where('id', $user->getKey()))
+                        ->orWhereHas('auditorTeams.users', fn (Builder $query) => $query->where('id', $user->getKey()))
+                        ->orWhereHas('managerUsers', fn (Builder $query) => $query->where('user_id', $user->getKey()))
+                        ->orWhereHas('auditorUsers', fn (Builder $query) => $query->where('user_id', $user->getKey()));
+                });
+            })
```

Comment:
After giving this some more thought. I think we should move this check for visibility off of the this list page and the `view` check of the policy and instead make it a global scope on the Model.

We still want to do the editing protections in the Policy. But access in general makes the most sense on the policy I think